httpie/manager/plugins.py

```diff
@@ -0,0 +1,188 @@
+import argparse
+import os
+import subprocess
+import sys
+import textwrap
+from collections import defaultdict
+from contextlib import suppress
+from pathlib import Path
+from typing import Optional, List
+
+import importlib_metadata
+
+from httpie.manager.cli import parser
+from httpie.compat import get_dist_name
+from httpie.context import Environment
+from httpie.status import ExitStatus
+
+
+class PluginInstaller:
+
+    def __init__(self, env: Environment, debug: bool = False) -> None:
+        self.env = env
+        self.dir = env.config.plugins_dir
+        self.debug = debug
+
+        self.setup_plugins_dir()
+
+    def setup_plugins_dir(self) -> None:
+        try:
+            self.dir.mkdir(
+                exist_ok=True,
+                parents=True
+            )
+        except OSError:
+            self.env.stderr.write(
+                f'Couldn\'t create "{self.dir!s}"'
+                ' directory for plugin installation.'
+                ' Please re-check the permissions for that directory,'
+                ' and if needed, allow write-access.'
+            )
+            raise
+
+    def fail(
+        self,
+        command: str,
+        target: Optional[str] = None,
+        reason: Optional[str] = None
+    ) -> ExitStatus:
+        message = f'Can\'t {command}'
+        if target:
+            message += f' {target!r}'
```

Comment:
why is `!r` needed? Isn't it already a string.

Reply:
`!r` calls `repr()` on the string, which makes it get printed with quotes. E.g:
```
>>> some_string = 's'
>>> print(f"{some_string}")
s
>>> print(f"{some_string!s}")
s
>>> print(f"{some_string!r}")
's'
```